text/0074-source-context-via-links.md

```diff
@@ -0,0 +1,70 @@
+- Start Date: 2023-02-15
+- RFC Type: feature
+- RFC PR: <https://github.com/getsentry/rfcs/pull/74>
+- RFC Status: draft
+
+# Summary
+
+In situations where a source code link and a line number is available for a frame, present the link in the UI and show
+the source code surrounding the line.
+
+# Motivation
+
+The source context is currently composed of three [`Frame` attributes](https://develop.sentry.dev/sdk/event-payloads/stacktrace/#frame-attributes):
+
+- `context_line` - Source code in filename at `lineno`.
+- `pre_context` - A list of source code lines before `context_line` (in order) – usually [`lineno - 5:lineno`].
+- `post_context` - A list of source code lines after `context_line` (in order) – usually [`lineno + 1:lineno + 5`].
+
+Showing a source code context and a link to it (e.g. to GitHub) currently works only if the source code itself is
+available during symbolication, thus the attributes are filled in at that time, or if they were already sent with the event by the SDK.
+
+There are, however, situations where we do have a URL where the source code resides, but not the contents (without downloading it), for example:
+
+- Portable-PDB source-link (.NET, see [this `symbolic` issue](https://github.com/getsentry/symbolic/issues/735))

```

Comment:
other examples include:

- Debuginfod servers (we don’t support these yet)
- SourceMaps (either embedded `sourcesContent` or using individual source files)

Reply:
```suggestion
- Portable-PDB source-link (.NET, see [this `symbolic` issue](https://github.com/getsentry/symbolic/issues/735))
- Debuginfod servers (we don’t support these yet)
- SourceMaps (either embedded sourcesContent or using individual source files)
```